src/extensions/g3d/model.ts

```diff
@@ -185,6 +202,19 @@ export function g3dHaplotypeQuery(haplotype: string) {
     });
 }
 
+export function g3dChromosomeQuery(chromosome: string) {
+    return MS.struct.generator.atomGroups({
+        'chain-test': MS.core.rel.eq([G3dSymbols.chromosome.symbol(), chromosome]),
+    });
+}
+
+export function g3dRegionQuery(chr: string, start: number, end: number) {
+    return MS.struct.generator.atomGroups({
+        'chain-test': MS.core.rel.eq([G3dSymbols.chromosome.symbol(), chr]),
```

Comment:
You can filter for sphere units in the chain test so the residue test would not be called needlessly (same for g3dChromosomeQuery to catch the edge case of an empty chain/chromosome name better). 

```ts
'chain-test': MS.core.logic.and([
    MS.core.rel.eq([MS.ammp('objectPrimitive'), 'sphere']),
    MS.core.rel.eq([G3dSymbols.chromosome.symbol(), chr])
])
```

Reply:
Thank you for comments @arose.  can you please give an example or modify the PR? I am not familiar with the mol* codebase, this was got from @dsehnal's example code before. good it's working :) Thank both of you!

Reply:
Well, just swap out the two 'chain-test's with the one I gave above and try to see if it still does what you want

Reply:
I see, thanks @arose, yes, it still works as what I want!